Melee Combat Events

OptimShi opened the pull request:
Added handlers for the following melee combat events:
ATTACKER_NOTIFICATION_EVENT
DEFENDER_NOTIFICATION_EVENT
EVASION_ATTACKER_NOTIFICATION_EVENT
EVASION_DEFENDER_NOTIFICATION_EVENT

tfarley commented on aclogview/CM_Combat.cs:
```diff
@@ -161,4 +185,121 @@ public class QueryHealthResponse : Message {
+            newObj.damage_type = binaryReader.ReadUInt32();
+            newObj.severity = binaryReader.ReadDouble();
+            newObj.damage = binaryReader.ReadUInt32();
+            newObj.critical = binaryReader.ReadBoolean();
```
From what I can see, it looks like critical is actually a full int, not just a single byte.

tfarley commented on aclogview/CM_Combat.cs:
```diff
@@ -161,4 +185,121 @@ public class QueryHealthResponse : Message {
+            newObj.severity = binaryReader.ReadDouble();
+            newObj.damage = binaryReader.ReadUInt32();
+            newObj.part = binaryReader.ReadUInt32();
+            newObj.critical = binaryReader.ReadBoolean();
```
Same, full int instead of bool for critical.

tfarley commented on aclogview/CM_Combat.cs:
```diff
@@ -161,4 +185,121 @@ public class QueryHealthResponse : Message {
+
+    public class DefenderNotificationEvent : Message
+    {
+        public PStringChar defenders_name;
```
This should be attackers_name

tfarley commented on aclogview/CM_Combat.cs:
```diff
@@ -161,4 +185,121 @@ public class QueryHealthResponse : Message {
+        {
+            TreeNode rootNode = new TreeNode(this.GetType().Name);
+            rootNode.Expand();
+            rootNode.Nodes.Add("defenders_name = " + defenders_name);
```
attackers_name

tfarley commented on aclogview/CM_Combat.cs:
```diff
@@ -161,4 +185,121 @@ public class QueryHealthResponse : Message {
+
+    public class EvasionAttackerNotificationEvent : Message
+    {
+        public PStringChar attacker;
```
Please make this defenders_name, just to be consistent

tfarley commented on aclogview/CM_Combat.cs:
```diff
@@ -161,4 +185,121 @@ public class QueryHealthResponse : Message {
+        {
+            TreeNode rootNode = new TreeNode(this.GetType().Name);
+            rootNode.Expand();
+            rootNode.Nodes.Add("attacker = " + attacker);
```
defenders_name

tfarley commented on aclogview/CM_Combat.cs:
```diff
@@ -161,4 +185,121 @@ public class QueryHealthResponse : Message {
+
+    public class EvasionDefenderNotificationEvent : Message
+    {
+        public PStringChar attacker;
```
Please make this attackers_name, just to be consistent

tfarley commented on aclogview/CM_Combat.cs:
```diff
@@ -161,4 +185,121 @@ public class QueryHealthResponse : Message {
+        {
+            TreeNode rootNode = new TreeNode(this.GetType().Name);
+            rootNode.Expand();
+            rootNode.Nodes.Add("defender = " + attacker);
```
attackers_name

tfarley commented on aclogview/Enums/Combat.cs:
```diff
@@ -111,3 +111,36 @@ public enum PowerBarMode {
+
+public enum BodyPart
+{
+    UNDEFINED,
```
These actually all need to shift by -1, so UNDEFINED = -1, HEAD = 0, CHEST = 1, etc. I would suggest just getting rid of the UNDEFINED entry.

tfarley commented on aclogview/Enums/Combat.cs:
```diff
@@ -111,3 +111,36 @@ public enum PowerBarMode {
+    SKIP, // Skip 12
+    FRONT_FOOT,
+    REAR_LEG,
+    SKIP_2, // Skip 15
```
After the above change, these two comments then become Skip 11 and Skip 14

OptimShi commented:
Thanks for taking the time to look at this and catch my mistakes! I've adjusted with your feedback and also added the enum for the "attack_conditions" bitfield (not currently reading them, but they are least there for reference for the time being)
